glommio/src/sys/uring.rs

```diff
@@ -879,32 +917,13 @@ impl UringCommon for PollRing {
     }
 
     fn submit_one_event(&mut self, queue: &mut VecDeque<UringDescriptor>) -> Option<bool> {
-        let source_map = &mut *self.source_map.borrow_mut();
-        let now = Instant::now();
-
-        while let Some(chain) = peek_one_chain(queue, self.size) {
-            return if let Some(sqes) = self.ring.sq().prepare_sqes(chain.len() as u32) {
-                let ops = extract_one_chain(source_map, queue, chain, now);
-                if ops.is_empty() {
-                    // all the sources in the ring were cancelled
-                    continue;
-                }
-
-                for (op, mut sqe) in ops.into_iter().zip(sqes.into_iter()) {
-                    let allocator = self.allocator.clone();
-                    fill_sqe(
-                        &mut sqe,
-                        &op,
-                        move |size| allocator.new_buffer(size),
-                        source_map,
-                    );
-                }
-                Some(true)
-            } else {
-                None
-            };
-        }
-        Some(false)
+        submit_event_chain(
+            &mut *self.source_map.borrow_mut(),
+            &mut self.ring,
+            self.allocator.clone(),
+            queue,
+            self.size,
+        )
     }
 }
```

Comment:
Looks like we could go one step further and remove `submit_one_event` from the trait and avoid the indirection of the vtable

Reply:
How do we do that? `submit_one_event` is used inside `submit_one_event` which is used in the default trait implementation where we cannot refer concrete fields like `self.ring`.